### ISSUE TITLE
Research: Learned semantic role labeling for entity disambiguation

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#578]
## Problem

While deterministic AST role analysis (#576) can distinguish grammatical roles (SUBJECT/OBJECT/MODIFIER), some cases require **semantic understanding**:

### Cases Requiring Semantic Disambiguation

1. **Metonymy**: "Esperanto kongresas en Parizo" - Esperanto (people) not language
2. **Ellipsis**: "La kreinto" - refers to Esperanto (language) but not mentioned
3. **Ambiguous compounds**: "Esperanto-literaturo" - is this modifier or entity itself?
4. **Contextual reference**: "ĝi" referring to Esperanto (language vs organization)

### Example Problem

**Document**: "La Esperanto-grupo en Berlino fondis novan sekcion."
- Grammatical role: "Esperanto" is MODIFIER
- Semantic meaning: Organization (not language)
- Query: "Kiu fondis sekcion?" → Should match (founding section)
- Query: "Kiu fondis Esperanton?" → Should NOT match (not founding language)

Current deterministic rules would penalize both queries equally.

### Architecture

Train small model (10-50M params) to classify entity mentions:

```python
class EntityMentionClassifier:
    """
    Input: entity mention + context (sentence AST)
    Output: semantic class
    
    Classes:
    - LANGUAGE: Esperanto as the language itself
    - ORGANIZATION: Esperanto-group, Esperanto-society
    - PERSON_COLLECTIVE: esperantistoj, Esperanto-movado
    - EVENT: Esperanto-kongreso
    - LITERATURE: Esperanto-literaturo, Esperanto-kurso
    - OTHER
    """
```

### Training Data Generation

**Weak supervision** from corpus patterns:
- "kreis/fondis Esperanton" → LANGUAGE
- "Esperanto-grupo/societo/asocio" → ORGANIZATION
- "esperantistoj/movado" → PERSON_COLLECTIVE
- "Esperanto-kongreso/kurso" → EVENT

Generate 10k-50k labeled examples from corpus.

### Integration with Retrieval

```python
# Query analysis
query_entity_type = classifier("Esperanton", query_ast)  # → LANGUAGE

# Document filtering
for doc in retrieved_docs:
    doc_entity_type = classifier("Esperanto", doc_ast)  # → ORGANIZATION
    
    if query_entity_type == doc_entity_type:
        score *= 2.0  # BOOST: semantic match
    else:
        score *= 0.5  # PENALIZE: semantic mismatch
```

## Research Questions

1. **Is this necessary?** Can deterministic rules + AST role analysis solve 90%+ cases?
2. **Training data quality**: Can weak supervision generate clean enough labels?
3. **Model size**: What's minimum model size for accurate classification?
4. **Generalization**: Does model overfit to training patterns?

### Step 1: Measure Deterministic Baseline
- Implement #576 (AST role analysis) and #577 (query understanding)
- Measure accuracy on test set
- Identify cases where deterministic approach fails

### Step 2: Error Analysis
- Manually annotate failure cases
- Categorize: grammatical ambiguity vs semantic ambiguity
- Estimate % solvable with learned model

### Step 3: Proof of Concept (if needed)
- Generate 1k weak-labeled examples
- Train small classifier (10M params)
- Measure improvement on failure cases

## Decision Criteria

**IMPLEMENT** learned model if:
- ✓ Deterministic baseline accuracy < 70%
- ✓ Semantic ambiguity causes >15% of failures
- ✓ Weak supervision generates >80% clean labels
- ✓ Model improves accuracy by >10 percentage points

**SKIP** learned model if:
- ✓ Deterministic baseline accuracy > 80%
- ✓ Most failures are data coverage issues, not semantic ambiguity
- ✓ Diminishing returns vs complexity cost

## Parent Epic

#574 - Grammatical Role-Based Entity Disambiguation

## Dependencies

- #576 - AST role analysis (baseline to beat)
- #577 - Query understanding (baseline to beat)
- #575 - Fix evaluation script (needed to measure improvement)

## Related Research

- #552 - Entity type classification (broader entity typing)
- #553 - Neural coreference resolution (related semantic understanding task)

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
